streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RankedClient.java

```diff
@@ -78,6 +79,28 @@ public int hashCode() {
         return Objects.hash(clientId, rank);
     }
 
+    /**
+     * Maps tasks to clients with caught-up states for the task.
+     *
+     * @param statefulTasksToRankedClients ranked clients map
+     * @return map from tasks with caught-up clients to the list of client candidates
+     */
+    static Map<TaskId, SortedSet<UUID>> tasksToCaughtUpClients(final SortedMap<TaskId, SortedSet<RankedClient>> statefulTasksToRankedClients) {
+        final Map<TaskId, SortedSet<UUID>> taskToCaughtUpClients = new HashMap<>();
+        for (final SortedMap.Entry<TaskId, SortedSet<RankedClient>> taskToRankedClients : statefulTasksToRankedClients.entrySet()) {
+            final SortedSet<RankedClient> rankedClients = taskToRankedClients.getValue();
+            for (final RankedClient rankedClient : rankedClients) {
+                if (rankedClient.rank() == Task.LATEST_OFFSET || rankedClient.rank() == 0) {
+                    final TaskId taskId = taskToRankedClients.getKey();
+                    taskToCaughtUpClients.computeIfAbsent(taskId, ignored -> new TreeSet<>()).add(rankedClient.clientId());
+                } else {
+                    break;
+                }
+            }
+        }
+        return taskToCaughtUpClients;
+    }
+
```

Comment:
req: Now that we can test this method. Could you please add unit tests for this method in `RankedClientTest`?